### PR TITLE
vilapuigvila array

### PR DESCRIPTION
**Pull request by vilapuigvila**
## Checklist

- [x] New Extension
- [x] New Test
- [ ] Changed more than one extension, but all changes are related
- [ ] Trivial change (doesn't require changelog)

### CODE REVIEW

File: EZSwiftExtensionsTests/ArrayTests.swift [hunk 1]
```diff
@@ -247,4 +247,14 @@ class ArrayTests: XCTestCase {
         XCTAssertEqual(duplicatedElementsArrayOrderMaintained.unique(), [2, 1, 3])
         
     }
+    
+    func testRemoveEqualItems() {
+        var list = ["one", "two", "two", "four"]
+        list.removeEqualItems(item: "two")
+        
+        XCTAssertEqual(["one", "four"], list)
```
Comment (Khalian) [inline EZSwiftExtensionsTests/ArrayTests.swift]: My first thought was that this API will remove duplicates of an item. It seems that is not true.

Could you refactor the name of this method to removeItemsWithValue?

File: Sources/ArrayExtensions.swift [hunk 2]
```diff
@@ -132,6 +132,13 @@ extension Array where Element: Equatable {
         self.remove(at: index)
     }
 
+    /// EZSE: Removes equal items of array
```
Comment (Khalian) [inline Sources/ArrayExtensions.swift]: Change the comment to removes all items in an array with a particular value.

File: EZSwiftExtensionsTests/ArrayTests.swift (hunk 1 above)
Comment (lfarah) [inline EZSwiftExtensionsTests/ArrayTests.swift]: @Khalian what do you think about making it more Swift 3?
``` swift
func removeEqualItems(with value: Element) {
    
}
```

Comment (Khalian) [inline EZSwiftExtensionsTests/ArrayTests.swift]: @lfarah 

   func removeAllItems(with value:Element) {

   }

is more appropriate

**Comment by piv199** [comment]
This is unnecessary extension, thus in swift we have: `array.filter { $0 != "two" }`, `array.filter { !["two", "three"].contains($0) }`

**Comment by Khalian** [comment]
@piv199 yea, most people do not understand anonymous methods though. 

Actually now I am kind of on the fence. @lfarah do you think this method adds value considering filters are trivial to implement in swift ?

**Comment by vilapuigvila** [comment]
@piv199 I'm agree, but if we use filter we need store filtered data in a new array, the original array still the same objects. My idea was to modify and work with one array. 
let newArray = list.filter({
    $0 != "two"
 })

File: Sources/ArrayExtensions.swift (hunk 2 above)
Comment (vilapuigvila) [inline Sources/ArrayExtensions.swift]: Ok, I'm going to do that. Should be a good idea to do your first thought?
 func removeRepeatedElemets()

**Comment by Khalian** [comment]
Cool. Now would you mind squashing these commits and adding an entry to the CHANGELOG.md for this?

**Comment by furuyan** [comment]
Could you tell me the difference with `func removeAll(_ elements: Element...)`?
It looks equal function.

**Comment by lfarah** [comment]
I do think this method adds value

File: EZSwiftExtensionsTests/ArrayTests.swift [hunk 3]
```diff
@@ -247,4 +247,14 @@ class ArrayTests: XCTestCase {
         XCTAssertEqual(duplicatedElementsArrayOrderMaintained.unique(), [2, 1, 3])
         
     }
+    
+    func testRemoveEqualItems() {
```
Comment (Khalian) [inline EZSwiftExtensionsTests/ArrayTests.swift]: Refactor name to testRemoveAllItemsWithValue.

**Comment by vilapuigvila** [comment]
@furuyan 👍  you're right is the same, so this PR can be closed, I'm sorry..

**Pull request closed without merging**